### PR TITLE
Apr 458 get organisation details for matched ukrpn

### PR DESCRIPTION
**Pull request by MarkFCain**

### CODE REVIEW

File: src/SFA.DAS.AssessorService.Application.Api/Services/UkrlpProcessingService.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            var providerDetail = new UkrlpProviderDetails();
+
+            var firstResult = providerDetails.ToList().FirstOrDefault();
+            providerDetail.LegalName = firstResult.ProviderName;
+            if (firstResult.ProviderAliases != null && firstResult.ProviderAliases.Count > 0)
+            {
+                providerDetail.TradingName = firstResult.ProviderAliases.First().Alias;
+            }
+
+            if (firstResult.VerificationDetails == null) return providerDetail;
```
Comment (David-B-Read) [inline src/SFA.DAS.AssessorService.Application.Api/Services/UkrlpProcessingService.cs]: Curly braces?

File: src/SFA.DAS.AssessorService.Application.Api/Services/UkrlpProcessingService.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+            foreach (var verificationDetail in firstResult.VerificationDetails)
+            {
+                if (verificationDetail.VerificationAuthority?.ToLower() == "companies house")
```
Comment (David-B-Read) [inline src/SFA.DAS.AssessorService.Application.Api/Services/UkrlpProcessingService.cs]: Just wondered if this, and the text searched for charity commission, should be constants, not string literals?

File: src/SFA.DAS.AssessorService.Web.Staff.Tests/Controllers/Roatp/AddRoatpOrganisationControllerTests.cs [hunk 3]
```diff
@@ -67,8 +78,8 @@ public void Add_organisation_details_initialises_with_list_of_organisation_types
             {
                 Errors = new List<ValidationErrorDetail>()
             };
-            _validator.Setup(x => x.ValidateOrganisationDetails(It.IsAny<AddOrganisationViewModel>()))
-                .ReturnsAsync(validationResult);
+            //_validator.Setup(x => x.ValidateOrganisationDetails(It.IsAny<AddOrganisationViewModel>()))
```
Comment (David-B-Read) [inline src/SFA.DAS.AssessorService.Web.Staff.Tests/Controllers/Roatp/AddRoatpOrganisationControllerTests.cs]: Remove comments

File: src/SFA.DAS.AssessorService.Application.Api/Controllers/UkrlpController.cs [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+            _ukrlpProcessingService = ukrlpProcessingService;
+        }
+
+        [HttpGet("provider-details/{ukprn}", Name = "GetProviderDetails")]
+        [SwaggerResponse((int)HttpStatusCode.OK, Type = typeof(UkrlpProviderDetails))]
+        [SwaggerResponse((int)HttpStatusCode.BadRequest, typeof(IDictionary<string, string>))]
+        [SwaggerResponse((int)HttpStatusCode.InternalServerError, Type = typeof(ApiResponse))]
+        public async Task<IActionResult> GetProviderDetails(long ukprn)
+        {
+            _logger.LogInformation("Get Provider details");
```
Comment (David-B-Read) [inline src/SFA.DAS.AssessorService.Application.Api/Controllers/UkrlpController.cs]: Should we include the UKPRN here - as we do further up the chain

File: src/SFA.DAS.AssessorService.Web.Staff.Tests/Controllers/Roatp/AddRoatpOrganisationControllerTests.cs [hunk 5]
```diff
@@ -16,29 +16,27 @@
...
         {
             _client = new Mock<IRoatpApiClient>();
-            _logger = new Mock<ILogger<AddRoatpOrganisationController>>();
-            _validator = new Mock<IAddOrganisationValidator>();
             _sessionService = new Mock<IRoatpSessionService>();
-            
-            _controller = new AddRoatpOrganisationController(_client.Object, _logger.Object, _validator.Object,
-                _sessionService.Object);
+            _ukrlpClient = new Mock<IUkrlpApiClient>()
+                ;
```
Comment (David-B-Read) [inline src/SFA.DAS.AssessorService.Web.Staff.Tests/Controllers/Roatp/AddRoatpOrganisationControllerTests.cs]: Remove whitespace

File: src/SFA.DAS.AssessorService.Web.Staff.Tests/Validators/AddOrganisatioViaUkprnViewModelValidatorTests.cs [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+using SFA.DAS.AssessorService.Api.Types.Models.Roatp;
+using SFA.DAS.AssessorService.Api.Types.Models.Validation;
+using SFA.DAS.AssessorService.Web.Staff.Infrastructure;
+using SFA.DAS.AssessorService.Web.Staff.Validators.Roatp;
+using SFA.DAS.AssessorService.Web.Staff.ViewModels.Roatp;
+
+namespace SFA.DAS.AssessorService.Web.Staff.Tests.Validators
+{
+    [TestFixture]
+    public class AddOrganisatioViaUkprnViewModelValidatorTests
```
Comment (David-B-Read) [inline src/SFA.DAS.AssessorService.Web.Staff.Tests/Validators/AddOrganisatioViaUkprnViewModelValidatorTests.cs]: Typo in classname

File: src/SFA.DAS.AssessorService.Web.Staff/Validators/Roatp/AddOrganisatioViaUkprnViewModelValidator.cs [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+using System.Threading.Tasks;
+using FluentValidation;
+using SFA.DAS.AssessorService.Api.Types.Models.Validation;
+using SFA.DAS.AssessorService.Web.Staff.Infrastructure;
+using SFA.DAS.AssessorService.Web.Staff.Resources;
+using SFA.DAS.AssessorService.Web.Staff.ViewModels.Roatp;
+
+namespace SFA.DAS.AssessorService.Web.Staff.Validators.Roatp
+{
+    public class AddOrganisatioViaUkprnViewModelValidator : AbstractValidator<AddOrganisatioViaUkprnViewModel>
```
Comment (David-B-Read) [inline src/SFA.DAS.AssessorService.Web.Staff/Validators/Roatp/AddOrganisatioViaUkprnViewModelValidator.cs]: Typo in class name

File: src/SFA.DAS.AssessorService.Web.Staff/ViewModels/Roatp/AddOrganisationViewModel.cs [hunk 8]
```diff
@@ -28,4 +28,20 @@ public AddOrganisationProviderTypeViewModel()
...
+
+    public class AddOrganisationTypeViewModel : AddOrganisationViewModel
+    {
+        public AddOrganisationTypeViewModel()
+        {
+            OrganisationId = Guid.NewGuid();
+        }
+    }
+
+    public class AddOrganisatioViaUkprnViewModel : AddOrganisationViewModel
```
Comment (David-B-Read) [inline src/SFA.DAS.AssessorService.Web.Staff/ViewModels/Roatp/AddOrganisationViewModel.cs]: Typo in class name

**Comment by davidgouge** [comment]
@alandburns The Uri in that json isn't the one that's used to connect to the service, but the one that was use to create the connection and generate the proxy classes.  It looks like the actual uri to connect to is set properly from config in Startup. 👍

**Pull request closed without merging**